include/common/mpmc_queue.h

```diff
@@ -56,9 +56,9 @@ class mpmc_queue{
         full{}
       {}
 
-      mpmc_queue(const mpmc_queue &) = delete;
+      mpmc_queue(const mpmc_queue & x) = delete; 
```

Comment:
Please. Do not add named parameter unless it is needed. In this case it is not needed.